### PR TITLE
Orion SLURM example

### PR DESCRIPTION
**Pull request by notoraptor**
Hello @lebrice ! This is my pull request to add an Orion minimalist SLURM example. It is intended to fix JIRA ticket https://mila-iqia.atlassian.net/browse/DOC-81 . It is based on existing example `distributed/001_single_gpu`.

What do you think?

PS:
* Orion indeed provides a command line `orion plot` to plot optimization result. However, it takes as argument the type of plot to provide (`lpi,parallel_coordinates,partial_dependencies,regret`), and I don't yet know which is the best plot to display. So, I have not added this command to the example, to avoid confusion. If necessary, I could ask @bouthilx for advice. What do you think?
* This is a screenshot for currently rendered documentation page. I don't know if example doc is placed in best position (bottom of page, and bottom in examples left menu). What do you think ?

https://mila-docs--198.org.readthedocs.build/en/198/Minimal_examples.html#hyperparameter-optimization-with-orion

### CODE REVIEW

File: docs/examples/frameworks/orion_setup/main.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+from torchvision.datasets import CIFAR10
+from torchvision.models import resnet18
+from tqdm import tqdm
+
+from orion.client import report_objective
+
+
+def main():
+    # Add an argument parser so that we can pass hyperparameters from command line.
+    parser = argparse.ArgumentParser(prog=sys.argv[0], description="Training parameters")
```
Comment (lebrice) [inline docs/examples/frameworks/orion_setup/main.py:23]:
```suggestion
    parser = argparse.ArgumentParser(description=__doc__)
```

File: docs/examples/frameworks/orion_setup/main.py [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+            progress_bar.set_postfix(loss=loss.item(), accuracy=accuracy.item())
+        progress_bar.close()
+
+        val_loss, val_accuracy = validation_loop(model, valid_dataloader, device)
+        logger.info(f"Epoch {epoch}: Val loss: {val_loss:.3f} accuracy: {val_accuracy:.2%}")
+        # Save latest accuracy
+        training_accuracy = val_accuracy.item()
+
+    # We report to Orion the objective that we want to minimize, which is 1 - training_accuracy.
+    report_objective(1 - training_accuracy)
```
[the remark this answers is not in the document]
Comment (lebrice) [reply docs/examples/frameworks/orion_setup/main.py:129]: You could also potentially report `val_loss` as the objective and `val_accuracy` as just a "metric" with Orion, I don't remember 100% how though.

**Comment by lebrice** [comment]
Actually, given the other tickets that we have left under https://mila-iqia.atlassian.net/browse/DOC-6, 
I think this could be nicely organized into a new "Running Hyper-Parameter Sweeps" / "Hyper-Parameter Optimization" / "Launching Multiple Jobs" section:

## Minimal Examples
  - Initial Setup
      - PyTorch
      - Jax
      - TensorFlow
      - MuJoCo
      - ...
  - Distributed Training
      - Single-GPU job
      - Multi-GPU job
      - Multi-Node job
  - Running Sweeps (this could be called differently) 
      - Grid search  (https://mila-iqia.atlassian.net/browse/DOC-80)
      - Hyper-Parameter Optimization with Orion (this example)
      - Hyper-Parameter Optimization with Wandb
      
Any thoughts @gyom @bouthilx  @obilaniu ?

File: docs/examples/frameworks/orion_setup/_index.rst [hunk 3]
```diff
@@ -0,0 +1,38 @@
+Orion example on single GPU Job
```
[the remark this answers is not in the document]
Comment (notoraptor) [reply docs/examples/frameworks/orion_setup/_index.rst]: Fait ici! https://github.com/mila-iqia/mila-docs/pull/198/commits/746d786f6f4dd5a9f435e1c1a53adf087e3d62d0

J'ai essayé d'écrire un paragraphe d'introduction mentionnant wandb comme autre framework du genre, je ne sais pas si c'est correct, ou s'il y en a d'autres.

File: docs/examples/frameworks/orion_setup/job.sh [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+# Execute Orion
+# Specify an experiment name with `-n`, which could be reused to display results (see below)
+# Specify max trials (here 10) to prevent a too-long run.
+# Then you can specify a search space for each `main.py`'s script parameter you want to optimize. 
+# Here we optimize only the learning rate.
+orion hunt -n orion-example --exp-max-trials 10 python main.py --learning-rate~'loguniform(1e-5, 1.0)'
+
+# Display Orion results
+orion info -n orion-example
```
[the remark this answers is not in the document]
Comment (notoraptor) [reply docs/examples/frameworks/orion_setup/job.sh]: Fait ici! https://github.com/mila-iqia/mila-docs/pull/198/commits/4dae317e44bb37b0074b6d7732da16be5d480fb7

File: docs/examples/frameworks/orion_setup/main.py [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+from orion.client import report_objective
+
+
+def main():
+    # Add an argument parser so that we can pass hyperparameters from command line.
+    parser = argparse.ArgumentParser(prog=sys.argv[0], description="Training parameters")
+    parser.add_argument('-e', '--epochs', type=int, default=10)
+    parser.add_argument('-l', '--learning-rate', type=float, default=5e-4)
+    parser.add_argument('-w', '--weight-decay', type=float, default=1e-4)
+    parser.add_argument('-n', '--batch-size', type=int, default=128)
```
[the remark this answers is not in the document]
Comment (notoraptor) [reply docs/examples/frameworks/orion_setup/main.py]: Fait ici! https://github.com/mila-iqia/mila-docs/pull/198/commits/98fb39c04f0fd1dab8663ee203f897955d3c0a38

File: docs/examples/frameworks/orion_setup/main.py (hunk 1 above)
Comment (notoraptor) [reply docs/examples/frameworks/orion_setup/main.py]: Fait! https://github.com/mila-iqia/mila-docs/pull/198/commits/b73df2ecd9c7f8392e86e9cb816370869f24272c

File: docs/examples/frameworks/orion_setup/main.py [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+    args = parser.parse_args(sys.argv[1:])
```
[the remark this answers is not in the document]
Comment (notoraptor) [reply docs/examples/frameworks/orion_setup/main.py]: Fait! https://github.com/mila-iqia/mila-docs/pull/198/commits/ddb6ee20373d5cc656ce1c56a64e58625321cb43

File: docs/examples/frameworks/orion_setup/main.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (notoraptor) [reply docs/examples/frameworks/orion_setup/main.py]: Fait! https://github.com/mila-iqia/mila-docs/pull/198/commits/7c5e1c662dbbb3d7e1e8fe249a42737857b98747

En effet, on peut aussi reporter l'accuracy en tant que statistique à Orion, mais j'ai seulement reporté le `val_loss` afin de garder le code simple.

**Comment by notoraptor** [comment]
@lebrice I addressed all your suggestions, except the latest comment about new examples organization. For this later one, I will wait further comments, maybe Monday?

**Comment by gyom** [comment]
I'm in favour of @lebrice's suggestion with a hierarchy something like this:

Running Sweeps (this could be called differently)
- Grid search (https://mila-iqia.atlassian.net/browse/DOC-80)
- Hyper-Parameter Optimization with Orion (this example)
- Hyper-Parameter Optimization with Wandb

I don't think the Orion example belongs under "distributed training".

File: docs/examples/frameworks/orion_setup/main.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (lebrice) [reply docs/examples/frameworks/orion_setup/main.py]: Dans ce commit-là il semblerait que tu report le training_loss

File: docs/examples/distributed/index.rst [hunk 7]
```diff
@@ -5,3 +5,4 @@ Distributed Training
 .. include:: /examples/distributed/001_single_gpu/_index.rst
 .. include:: /examples/distributed/002_multi_gpu/_index.rst
 .. include:: /examples/distributed/003_multi_node/_index.rst
+.. include:: /examples/frameworks/orion_setup/_index.rst
```
Comment (lebrice) [inline docs/examples/distributed/index.rst:8]: Comme mentionné auparavant, j'inclurais plutot l'exemple dans une nouvelle section, appellée "Running Sweeps", au même niveau que "Distributed Training"

File: docs/examples/frameworks/orion_setup/main.py [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+
+            # Advance the progress bar one step, and update the "postfix" () the progress bar. (nicer than just)
+            progress_bar.update(1)
+            progress_bar.set_postfix(loss=loss.item(), accuracy=accuracy.item())
+        progress_bar.close()
+
+        val_loss, val_accuracy = validation_loop(model, valid_dataloader, device)
+        logger.info(f"Epoch {epoch}: Val loss: {val_loss:.3f} accuracy: {val_accuracy:.2%}")
+        # Save latest val_loss
+        training_loss = val_loss
```
Comment (lebrice) [inline docs/examples/frameworks/orion_setup/main.py:126]:
```suggestion
```

File: docs/examples/frameworks/orion_setup/main.py [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+
+    # We report to Orion the objective that we want to minimize, which is latest val_loss.
+    report_objective(training_loss)
```
Comment (lebrice) [inline docs/examples/frameworks/orion_setup/main.py:129]:
```suggestion
    report_objective(val_loss)
```

**Review by lebrice — CHANGES_REQUESTED** [review]

File: docs/examples/frameworks/orion_setup/main.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (notoraptor) [reply docs/examples/frameworks/orion_setup/main.py]: Salut, Fabrice ! Le `training_loss` est juste une variable qui stocke le dernier `val_loss` de la boucle des epochs.

Si tu veux, je peux directement reporter `val_loss`. C'est juste qu'il faut être sûr que le script passe dans la boucle au moins 1 fois, pour que la variable `val_loss` existe. Mais peut-être que je n'ai pas besoin de me soucier de tout ça, vu que ce n'est qu'un code d'exemple ?

[the remark this answers is not in the document]
Comment (notoraptor) [reply docs/examples/frameworks/orion_setup/main.py]: PS: j'ai vu tes commentaires ci-dessous, donc c'est fait! https://github.com/mila-iqia/mila-docs/pull/198/commits/d4c102a0b460fb34838adb7a75b12d2eae8d7ce2

File: docs/examples/frameworks/orion_setup/main.py (hunk 8 above)
Comment (notoraptor) [reply docs/examples/frameworks/orion_setup/main.py]: Fait! https://github.com/mila-iqia/mila-docs/pull/198/commits/d4c102a0b460fb34838adb7a75b12d2eae8d7ce2

File: docs/examples/frameworks/orion_setup/main.py (hunk 9 above)
Comment (notoraptor) [reply docs/examples/frameworks/orion_setup/main.py]: Fait! https://github.com/mila-iqia/mila-docs/pull/198/commits/d4c102a0b460fb34838adb7a75b12d2eae8d7ce2

File: docs/examples/distributed/index.rst (hunk 7 above)
Comment (notoraptor) [reply docs/examples/distributed/index.rst]: Fait! https://github.com/mila-iqia/mila-docs/pull/198/commits/b9b471f753163fb4ca00de4bc87ad50b3a94352e

J'ai seulement rajouté la section `Running Sweeps` avec la sous-sections relative à Orion. Je n'ai pas ajouté les autres éléments dont vous parliez car je me suis dit que ça ne relevait pas de cette PR-ci.

**Comment by notoraptor** [comment]
@lebrice Voici une capture d'écran du rendu actuel de la doc.

Screenshot 2023-06-26 at 13-30-41 Software Frameworks — MILA Technical Documentation latest documentation

**Comment by notoraptor** [comment]
@gyom Fait ! https://github.com/mila-iqia/mila-docs/pull/198/commits/749c42230a56e84fcde64ceb7613eb5a010bc73f

Screenshot 2023-06-26 at 13-52-18 Software Frameworks — MILA Technical Documentation latest documentation

**Comment by gyom** [comment]
Même la documentation d'Orion elle-même écrit le mot "hyperparameter" en un seul mot. Je pense qu'on devrait écrire ça comme ça aussi. Wikipédia l'écrit en un mot aussi. La seule raison pourquoi on voudrait le casser en deux, ça serait si on voulait vraaaaiment rendre ça évident pour qqun pourquoi on écrit l'acronyme "HPO" entre parenthèses après.

**Comment by notoraptor** [comment]
@gyom Fait! https://github.com/mila-iqia/mila-docs/pull/198/commits/5425d5cfaab48fa32493ba66a99bf1f5b6746e14

Screenshot 2023-06-26 at 14-05-03 Software Frameworks — MILA Technical Documentation latest documentation

**Comment by bouthilx** [comment]
J'ai horreur du terme 'sweep'. Ça a juste rapport avec un grid search, et à la limite un random search, mais sinon aucunement avec de l'optimization.

Je crois que l'example lié à ce qu'Olexa a fait c'est plus du 'manual optimization' que du grid search. Si on ajoute un example pour le grid search spécifiquement, on devrait au minimum expliquer dans quel situation c'est une bonne idée et sinon diriger vers un autre algorithme. Je ne pourrai pas dormir en paix le soir si on documente grid-search sans avertir contre sa mauvaise utilisation. :sweat_smile:

**Review by lebrice — APPROVED** [review]

**Comment by lebrice** [comment]
Hey @bouthilx on a besoin d'un autre review pour merge, si jamais tu as quelques minutes

File: docs/examples/hpo/hpo_with_orion/README.rst [hunk 10]
```diff
@@ -0,0 +1,324 @@
...
+    module load anaconda/3
+    module load cuda/11.7
+
+    # Creating the environment for the first time:
+    # conda create -y -n pytorch python=3.9 pytorch torchvision torchaudio \
+    #     pytorch-cuda=11.7 -c pytorch -c nvidia
+    # Other conda packages:
+    # conda install -y -n pytorch -c conda-forge rich tqdm
+   +# Orion package:
+   +# pip install orion[profet]
```
Comment (bouthilx) [inline docs/examples/hpo/hpo_with_orion/README.rst:52]: Pourquoi on installe avec `profet`?

File: docs/examples/hpo/hpo_with_orion/README.rst [hunk 11]
```diff
@@ -0,0 +1,324 @@
...
+                # Advance the progress bar one step, and update the "postfix" () the progress bar. (nicer than just)
+                progress_bar.update(1)
+                progress_bar.set_postfix(loss=loss.item(), accuracy=accuracy.item())
+            progress_bar.close()
+
+            val_loss, val_accuracy = validation_loop(model, valid_dataloader, device)
+            logger.info(f"Epoch {epoch}: Val loss: {val_loss:.3f} accuracy: {val_accuracy:.2%}")
+
+   +    # We report to Orion the objective that we want to minimize, which is latest val_loss.
+   +    report_objective(val_loss)
```
Comment (bouthilx) [inline docs/examples/hpo/hpo_with_orion/README.rst:224]:
```suggestion
   +    report_objective(val_accuracy)
```

Les HP devraient être optimisé avec l'accuracy. Le _loss_ c'est juste un _surrogate_ pour le vrai _risk_ qu'on veut minimiser, qui lui est l'erreur de classification (1 - accuracy).

Comment (lebrice) [reply docs/examples/hpo/hpo_with_orion/README.rst:224]: On devrait pas alors faire `report_objective(1 - accuracy)` plutot que `accuracy`?

File: docs/examples/hpo/hpo_with_orion/README.rst [hunk 12]
```diff
@@ -0,0 +1,324 @@
+Hyperparameter Optimization with Oríon
```
Comment (bouthilx) [inline docs/examples/hpo/hpo_with_orion/README.rst:1]: À mon avis l'exemple devrait mentionner qu'on peut facilement paralleliser en lançant plusieurs jobs en même temps. On a ici un exemple qui montre comment l'utiliser avec slurm et l'option `--array`: https://orion.readthedocs.io/en/stable/tutorials/cluster.html#parallel-optimization-using-arrays

File: docs/examples/hpo/hpo_with_orion/README.rst (hunk 11 above)
Comment (bouthilx) [reply docs/examples/hpo/hpo_with_orion/README.rst:224]: En effet, tu as dû voir la première version que j'ai envoyé avant de corriger. :sweat_smile:

**Comment by bouthilx** [comment]
> Orion indeed provides a command line orion plot to plot optimization result. However, it takes as argument the type of plot to provide (lpi,parallel_coordinates,partial_dependencies,regret), and I don't yet know which is the best plot to display. So, I have not added this command to the example, to avoid confusion. If necessary, I could ask @bouthilx for advice. What do you think?

On pourrait mettre le `regret` qui est simple et ne demande pas trop d'explications, et référer à https://orion.readthedocs.io/en/stable/auto_examples/plot_4_partial_dependencies.html pour un plot plus complexe mais plus utile.

**Comment by satyaog** [comment]
Btw, if you click on Details next to the readthedocs check you land on the current rendering of the doc, in case you would prefer not to do screenshots :

https://mila-docs--198.org.readthedocs.build/en/198/

image

File: docs/examples/hpo/hpo_with_orion/README.rst (hunk 12 above)
Comment (lebrice) [reply docs/examples/hpo/hpo_with_orion/README.rst:1]: Dans l'ordre, ce serait préférable de montrer aux usagers comment lancer plusieurs tâches dans une même job, avant de les inciter à scaler avec les job arrays.

Comment (bouthilx) [reply docs/examples/hpo/hpo_with_orion/README.rst:1]: Des tâches parallels? J'ai l'impression que c'est plus simple si chaque tâche est dans une job séparé.

Comment (lebrice) [reply docs/examples/hpo/hpo_with_orion/README.rst:1]: Ouais, mais on veut idéallement qu'ils utilisent le GPU au complet avec plusieurs runs en parallèle avant de lancer plein de jobs. Jsuis d'accord que c'est vraiment plus facile de scaler avec des job arrays.

Comment (lebrice) [reply docs/examples/hpo/hpo_with_orion/README.rst:1]: Dans ce cas-ci, ça utilise 1.5G de VRAM, donc sur une RTX8000 on peut en principe en fitter 30 runs en parallèle!

**Comment by lebrice** [comment]
Je suggère qu'on déplace cet exemple sous la nouvelle section "Good practices" @notoraptor.

File: docs/examples/hpo/hpo_with_orion/README.rst (hunk 10 above)
Comment (notoraptor) [reply docs/examples/hpo/hpo_with_orion/README.rst:52]: Salut @bouthilx ! Si j'installe orion seul (`pip install orion`), le script actuel demandera que profet soit aussi installé:

```
/home/mila/b/boccos/.conda/envs/pytorch/lib/python3.9/site-packages/orion/benchmark/task/profet/model_utils.py:32: RuntimeWarning: The `profet` extras needs to be installed in order to use the Profet tasks.
Error: No module named 'GPy'
Use `pip install orion[profet]` to install the profet extras.
```

Comment (notoraptor) [reply docs/examples/hpo/hpo_with_orion/README.rst]: PS: Tout compte fait, le script roule en effet jusqu'au bout même si profet n'est pas installé. Par contre, ça affichera quand même le message d'erreur/warning suggérant à l'utilisateur d'installer profet.

Comme ça fonctionne sans profet, j'ai changé la consigne en mettant juste `pip install orion`: https://github.com/mila-iqia/mila-docs/pull/198/commits/bc2d3392b63321931f4162bd26f9e908580d2a12

File: docs/examples/hpo/hpo_with_orion/README.rst (hunk 11 above)
Comment (notoraptor) [reply docs/examples/hpo/hpo_with_orion/README.rst]: Corrigé: https://github.com/mila-iqia/mila-docs/pull/198/commits/bc2d3392b63321931f4162bd26f9e908580d2a12

File: docs/examples/hpo/hpo_with_orion/README.rst (hunk 12 above)
Comment (notoraptor) [reply docs/examples/hpo/hpo_with_orion/README.rst:1]: Du coup, ici, je ne change rien ?

Comment (bouthilx) [reply docs/examples/hpo/hpo_with_orion/README.rst:1]: 30! :scream: wow

Ce serait bien que les deux soit présent. Sinon ça ne mentionne nulle part comment scaler à travers plusieurs jobs.

Comment (lebrice) [reply docs/examples/hpo/hpo_with_orion/README.rst:1]: Le scaling avec plusieurs runs dans la même job c'est Simon-Olivier qui s'en charge, donc je propose qu'on garde ça comme ça pour l'instant en effet @notoraptor

**Comment by notoraptor** [comment]
> > Orion indeed provides a command line orion plot to plot optimization result. However, it takes as argument the type of plot to provide (lpi,parallel_coordinates,partial_dependencies,regret), and I don't yet know which is the best plot to display. So, I have not added this command to the example, to avoid confusion. If necessary, I could ask @bouthilx for advice. What do you think?
> 
> On pourrait mettre le `regret` qui est simple et ne demande pas trop d'explications, et référer à https://orion.readthedocs.io/en/stable/auto_examples/plot_4_partial_dependencies.html pour un plot plus complexe mais plus utile.

Fait: https://github.com/mila-iqia/mila-docs/pull/198/commits/cdb712a9fe0d8eeeed82c7803662ef2a2b7c79d5

File: docs/examples/hpo/hpo_with_orion/README.rst (hunk 12 above)
Comment (bouthilx) [reply docs/examples/hpo/hpo_with_orion/README.rst:1]: Je ne comprends pas, il n'y a aucune mention ici sur comment rouler plus qu'un entrainement du HPO en parallel non? Il devrait y avoir au minimum une mention sur comment le faire avec plusieurs jobs (juste ajouter --array) parce que du HPO séquentiel c'est vraiment pas pratique.

Est-ce que l'example de Simon-Olivier implique du HPO?

**Comment by notoraptor** [comment]
Code rebasé.

@lebrice Concernant le déplacement de la section sous "Good practices", comment dois-je le faire ? Actuellement, ça ressemble à ceci:

Screenshot 2023-07-10 at 15-49-01 Software Frameworks — MILA Technical Documentation latest documentation

Est-ce que:
- Je laisse ça comme ça ?
- Je déplace la section "hyperparameter Optimization" à l'intérieur de la section "Good practices" ?
- Je déplace seulement la sous-section "hyperparameter optimization with orion" à l'intérieur de la section "Good practices", et donc je supprime la section "hyperparameter optimization" ?

**Comment by notoraptor** [comment]
@satyaog merci pour l'info sur la doc ! En effet, c'est plus simple de la consulter depuis le Github Actions, au lieu de générer une longue image chaque fois.

**Comment by lebrice** [comment]
> Je déplace seulement la sous-section "hyperparameter optimization with orion" à l'intérieur de la section "Good practices", et donc je supprime la section "hyperparameter optimization" ?

Ceci, puisqu'on a pas encore d'autres pages à mettre dans cette section là. Si jamais on en rajoute, on pourra penser à en faire une section.

File: docs/examples/good_practices/hpo_with_orion/README.rst [hunk 13]
```diff
@@ -0,0 +1,333 @@
...
+<https://orion.readthedocs.io/en/stable/index.html>`_.
+
+
+**job.sh**
+
+.. code:: diff
+
+    # distributed/001_single_gpu/job.sh -> good_practices/hpo_with_orion/job.sh
+   old mode 100644
+   new mode 100755
```
Comment (satyaog) [inline docs/examples/good_practices/hpo_with_orion/README.rst:25]: this should probably not be there right?

File: docs/examples/good_practices/hpo_with_orion/README.rst [hunk 14]
```diff
@@ -0,0 +1,333 @@
...
+        # Create a model and move it to the GPU.
+        model = resnet18(num_classes=10)
+        model.to(device=device)
+
+        optimizer = torch.optim.AdamW(model.parameters(), lr=learning_rate, weight_decay=weight_decay)
+
+        # Setup CIFAR10
+        num_workers = get_num_workers()
+   -    dataset_path = Path(os.environ.get("SLURM_TMPDIR", ".")) / "data"
+   +    dataset_path = Path(os.environ.get("SLURM_TMPDIR", "")) / "data"
```
Comment (satyaog) [inline docs/examples/good_practices/hpo_with_orion/README.rst:156]: Ain't this change unnecessary and should be done instead in the base example?

File: docs/examples/good_practices/hpo_with_orion/main.py [hunk 15]
```diff
@@ -0,0 +1,193 @@
...
+
+    # Create a model and move it to the GPU.
+    model = resnet18(num_classes=10)
+    model.to(device=device)
+
+    optimizer = torch.optim.AdamW(model.parameters(), lr=learning_rate, weight_decay=weight_decay)
+
+    # Setup CIFAR10
+    num_workers = get_num_workers()
+    dataset_path = Path(os.environ.get("SLURM_TMPDIR", "")) / "data"
```
Comment (lebrice) [inline docs/examples/good_practices/hpo_with_orion/main.py:57]:
```suggestion
    dataset_path = Path(os.environ.get("SLURM_TMPDIR", ".")) / "data"
```

File: docs/examples/good_practices/hpo_with_orion/README.rst (hunk 13 above)
Comment (notoraptor) [reply docs/examples/good_practices/hpo_with_orion/README.rst]: En effet, corrigé! https://github.com/mila-iqia/mila-docs/pull/198/commits/2a69570d1e6da409b4fe4993310aee1df1ef0305

File: docs/examples/good_practices/hpo_with_orion/main.py (hunk 15 above)
Comment (notoraptor) [reply docs/examples/good_practices/hpo_with_orion/main.py]: Fait! https://github.com/mila-iqia/mila-docs/pull/198/commits/4a3b3ff1c0892ddd28f10e0e06c2e0bae290fdf0

File: docs/examples/good_practices/hpo_with_orion/README.rst (hunk 14 above)
Comment (notoraptor) [reply docs/examples/good_practices/hpo_with_orion/README.rst]: En effet, corrigé dans le dernier commit! https://github.com/mila-iqia/mila-docs/pull/198/commits/4a3b3ff1c0892ddd28f10e0e06c2e0bae290fdf0

**Comment by notoraptor** [comment]
PS: `build_the_docs_test` échoue, mais je ne comprends pas pourquoi.

**Comment by satyaog** [comment]
btd should be fixed for now by #204 . Rebase this PR on master and it should be good

**Comment by notoraptor** [comment]
@satyaog Indeed, rebased, and tests passed.

**Review by satyaog — APPROVED** [review]

**Pull request merged**